cmd/commands/deploy.go

```diff
@@ -72,10 +72,30 @@ func deploy(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	dsnToKeySteps := make(map[string]string)
+
+	for key, step := range fn.Steps {
+		dsnToKeySteps[step.DSN(ctx, *fn)] = key
+	}
+
 	for _, a := range actions {
-		if err := deployAction(ctx, a); err != nil {
+		actionVersion, err := deployAction(ctx, a)
+		if err != nil {
 			return err
 		}
+
+		// TODO: Move this to a dedicated function.
+		step, ok := fn.Steps[dsnToKeySteps[actionVersion.DSN]]
```

Comment:
I found another spot in the code that matches a `Step` to an `ActionVersion` via the `DSN`.

I wanted to add a `func (a ActionVersion) Step(steps []Step) *Step` but ran into a circular dependency.

I'd love to quickly spike with someone about the best way to alleviate those issues in the future. My guess is moving packages around, which feels unideal.

Reply:
Yeah, this is part of our cleanup we need to do with the top-level `inngest` package, plus actions in general.  Definitely agree on tidying this in the future.